### PR TITLE
[typespec-next] React to azure/typespec-azure#3180

### PR DESCRIPTION
**Pull request by mikeharder**
This PR includes changes reacting to the fix for Azure/typespec-azure#3180

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp [hunk 1]
```diff
@@ -189,14 +189,12 @@ model Choice {
   index: int32;
 
   #suppress "@azure-tools/typespec-azure-core/no-nullable" "The operation already returns nulls"
-  #suppress "@azure-tools/typespec-autorest/union-unsupported" "https://github.com/Azure/typespec-azure/issues/3180"
```
Comment (mikeharder) [inline specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp:192]: One thing a little strange, is I was able to remove the `#suppress` for `CompletionsLogProbabilityModel`, but there is no change for this type in the autorest (only for `CompletionsFinishReason`).  Is this OK?

Comment (timotheeguerin) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp:192]: I think that suppress was incorrect, there wasn't an error reported here. 

`CompletionsLogProbabilityModel` is a model

**Review by timotheeguerin — APPROVED** [review]

File: specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp (hunk 1 above)
Comment (mikeharder) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp:192]: I will try to remove this suppress in main

Comment (mikeharder) [reply specification/cognitiveservices/OpenAI.Inference/models/completions.create.tsp:192]: Removing this in main passes CI, so it must have unnecessarily been added in error.

https://github.com/Azure/azure-rest-api-specs/pull/24562

**Pull request merged**